src/main/java/com/glodblock/github/network/CPacketRenamer.java

```diff
@@ -97,11 +97,15 @@ public void toBytes(ByteBuf buf) {
 
     public static class Handler implements IMessageHandler<CPacketRenamer, IMessage> {
 
+        private String getName(ICustomNameObject obj) {
+            return obj.hasCustomName() ? obj.getCustomName() : "";
+        }
+
         private String getName(TileEntity tile, ForgeDirection side) {
             if (tile instanceof TileCableBus) {
-                return ((ICustomNameObject) ((TileCableBus) tile).getPart(side)).getCustomName();
+                return getName((ICustomNameObject) ((TileCableBus) tile).getPart(side));
             } else {
-                return ((ICustomNameObject) tile).getCustomName();
+                return getName((ICustomNameObject) tile);
```

Comment:
Is this a safe cast?

Reply:
Isn't this inside CPacketRenamer which would only be relevant in the first place when it's an `ICustomNameObject`?

Reply:
This `CPacketRenamer` controls the rename button, which show up when you hold shift in the wireless pattern terminal, and does not exist in the wired pattern terminal. 

There is a potential bug here, regardless of whether a tile impliment `ICustomNameObject` or not, currently the button still changes, and the request is sent without checking if `ICustomNameObject` is implemented. So this might be a problem if you add tiles that is a "interface", but does not implement `ICustomNameObject`. 

https://github.com/GTNewHorizons/AE2FluidCraft-Rework/blob/f1cd3b1afcb3ce95a73a59a008e5a60f7b52845b/src/main/java/com/glodblock/github/client/gui/GuiInterfaceTerminalWireless.java#L323-L327

https://github.com/GTNewHorizons/AE2FluidCraft-Rework/blob/f1cd3b1afcb3ce95a73a59a008e5a60f7b52845b/src/main/java/com/glodblock/github/client/gui/GuiInterfaceTerminalWireless.java#L364-L371

It's not a problem now because all the "interfaces" implement it, and future integrations really should implement it, might as well wait until there's a reason for unnamed interface to exist before fixing it, which I doubt there is.

Reply:
This is how we get erroneous crashes in the future and a brittle codebase, because we decided that at the time it was fine, but later on we want to add functionality contrary to the expectations laid out before